### PR TITLE
chore: update cid/multihash

### PR DESCRIPTION
**Pull request by Stebalien**
Pull in a large breaking change to cid & multihash, updating the FVM to handle these changes as well.

### CODE REVIEW

**Comment by Stebalien** [comment]
Hm. That test failure is interesting.

**Comment by Stebalien** [comment]
Ah, it was a bug fix.

File: testing/integration/Cargo.toml [hunk 1]
```diff
@@ -23,6 +23,6 @@ serde = { workspace = true }
 serde_tuple = { workspace = true }
 
 [dev-dependencies]
-actors-v12 = { package = "fil_builtin_actors_bundle", git = "https://github.com/filecoin-project/builtin-actors", branch = "fvm-next" }
+actors = { package = "fil_builtin_actors_bundle", git = "https://github.com/filecoin-project/builtin-actors", branch = "steb/update-multihash" }
```
[the remark this answers is not in the document]
Comment (Stebalien) [reply testing/integration/Cargo.toml:26]: Specifically, the plan is:

1. Merge this PR.
2. Release these crates.
3. Update these crates in the builtin actor's PR (https://github.com/filecoin-project/builtin-actors/pull/1589).
4. Merge that PR.
5. Create a new PR here to switch the builtin actors dependency to the master branch.

**Review by ZenGround0 — APPROVED** [review]

File: testing/integration/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (rvagg) [reply testing/integration/Cargo.toml:26]: and actors-utils too ...

**Review by rvagg — APPROVED** [review]

File: testing/integration/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Stebalien) [reply testing/integration/Cargo.toml:26]: Well, technically this is just a dev dependency so re-releasing these crates isn't really necessary.

**Pull request merged**